### PR TITLE
[CocoaPods] Re-work how privacy manifests are exposed

### PR DESCRIPTION
**Pull request by ncooke3**
Follow-up to add privacy manifests to #35042

- [x] Update podspec template files
- [x] Align on manifest update:
```
git grep mach_absolute_time
src/core/lib/gpr/posix/time.cc:static uint64_t g_time_start = mach_absolute_time();
src/core/lib/gpr/posix/time.cc:                ((double)(mach_absolute_time() - g_time_start)) * g_time_scale;
```

cc: @paulb777

### CODE REVIEW

**Comment by sampajano** [comment]
@HannahShiSFB Hey Hannah, could you help take a look at this PR on helping us improve the privacy manifest? This is important because Apple's requirement date is coming soon.

**Comment by HannahShiSFB** [comment]
@ncooke3 Were you able to generate the privacy report?
I finally fixed the signing issue and was able to generate a privacy report but it was a 807 bytes blank pdf.

**Comment by ncooke3** [comment]
> @ncooke3 Were you able to generate the privacy report? I finally fixed the signing issue and was able to generate a privacy report but it was a 807 bytes blank pdf.

Hi @HannahShiSFB, thanks for working on the testing. I believe the empty Privacy Report is this working as intended. Did you notice the Privacy Report was non-empty when testing #35042?

My understanding is that because the Privacy Manifest has an empty data collection disclosure (code block below), the App Privacy Report will have no entries when the app depends on any of the  gRPC pods touched by this PR.

```
<key>NSPrivacyCollectedDataTypes</key>
	<array/>
```

In the experimenting I've done, the required reasons section of the Privacy Manifest does not seem to show in the Privacy Report. 

If you inspect the app bundle (I usually run `tree /path/to/MyApp.app`), you should see a bundle containing a privacy manifest for each gRPC pod the app depends on.

**Review by HannahShiSFB — APPROVED** [review]

**Comment by HannahShiSFB** [comment]
Thanks for the change, I updated the manifest with some NSPrivacyCollectedDataTypes, now I can see them in the reports
[SwiftUseFrameworks-PrivacyReport 2024-01-20 09-57-24.pdf](https://github.com/grpc/grpc/files/13998791/SwiftUseFrameworks-PrivacyReport.2024-01-20.09-57-24.pdf)

I can see it in grpc, grpc++ and grpc client, but not in boring ssl. tree command also indicates the PrivacyInfo.xcprivacy file is not added.

File: templates/src/objective-c/BoringSSL-GRPC.podspec.template [hunk 1]
```diff
@@ -152,6 +152,11 @@
     end
     s.subspec 'Implementation' do |ss|
       ss.header_mappings_dir = 'src'
+
+      ss.resource_bundles = {
+        s.module_name => 'src/objective-c/PrivacyInfo.xcprivacy'
```
Comment (HannahShiSFB) [inline templates/src/objective-c/BoringSSL-GRPC.podspec.template:157]: This podspec is already in 'src/objective-c/' directory, so should be
`s.module_name => 'PrivacyInfo.xcprivacy'`
here.

Comment (ncooke3) [reply templates/src/objective-c/BoringSSL-GRPC.podspec.template:157]: Good catch, @HannahShiSFB!

**Pull request closed without merging**